core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceSettingService.scala

```diff
@@ -53,32 +57,40 @@ class WorkspaceSettingService(protected val ctx: RawlsRequestContext,
-        case WorkspaceSetting(settingType @ WorkspaceSettingTypes.GcpBucketLifecycle,
-                              GcpBucketLifecycleConfig(rules)
-            ) =>
-          rules.flatMap { rule =>
-            val actionValidation = rule.action.actionType match {
-              case actionType if actionType.equals("Delete") => None
-              case actionType => Some(validationErrorReport(settingType, s"unsupported lifecycle action $actionType"))
-            }
-            val ageValidation = rule.conditions.age.collect {
-              case age if age < 0 =>
-                validationErrorReport(settingType, "age must be a non-negative integer")
-            }
-            val atLeastOneConditionValidation = rule.conditions match {
-              case GcpBucketLifecycleCondition(None, None) =>
-                Some(validationErrorReport(settingType, "at least one condition must be specified"))
-              case GcpBucketLifecycleCondition(Some(prefixes), None) if prefixes.isEmpty =>
-                Some(
-                  validationErrorReport(settingType,
-                                        "at least one prefix must be specified if matchesPrefix is the only condition"
+      val validationErrors = requestedSettings.flatMap { setting =>
+        setting match {
+          case GcpBucketLifecycleSetting(GcpBucketLifecycleConfig(rules)) =>
+            rules.flatMap { rule =>
+              val actionValidation = rule.action.actionType match {
+                case actionType if actionType.equals("Delete") => None
+                case actionType =>
+                  Some(validationErrorReport(setting.settingType, s"unsupported lifecycle action $actionType"))
+              }
+              val ageValidation = rule.conditions.age.collect {
+                case age if age < 0 =>
+                  validationErrorReport(setting.settingType, "age must be a non-negative integer")
+              }
+              val atLeastOneConditionValidation = rule.conditions match {
+                case GcpBucketLifecycleCondition(None, None) =>
+                  Some(validationErrorReport(setting.settingType, "at least one condition must be specified"))
+                case GcpBucketLifecycleCondition(Some(prefixes), None) if prefixes.isEmpty =>
+                  Some(
+                    validationErrorReport(setting.settingType,
+                                          "at least one prefix must be specified if matchesPrefix is the only condition"
+                    )
                   )
-                )
+                case _ => None
+              }
+              actionValidation ++ ageValidation ++ atLeastOneConditionValidation
+            }
+          case GcpBucketSoftDeleteSetting(GcpBucketSoftDeleteConfig(retentionDuration)) =>
+            val retentionDurationValidation = retentionDuration match {
+              case retentionDuration if retentionDuration < 0 || retentionDuration > 7_776_000 =>
```

Comment:
[Magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) alert, consider extracting this to a named constant.